kettle-plugins/hdfs/src/main/java/org/pentaho/big/data/kettle/plugins/hdfs/trans/HadoopFileOutputMeta.java

```diff
@@ -105,9 +111,27 @@ protected String getProcessedUrl( IMetaStore metastore, String url ) {
     if ( url == null ) {
       return null;
     }
-    NamedCluster c = namedClusterService.getNamedClusterByName( sourceConfigurationName, metastore );
+    if ( metastore == null ) {
+      // Maybe we can get a metastore from spoon
+      if ( Spoon.getInstance() != null ) {
```

Comment:
This still creeps me out. We don't always have the UI jar. It may be "working" now since Pan does have kettle-ui, but this is going to break at some point.
Use this instead:
org.pentaho.di.metastore.MetaStoreConst#openLocalPentahoMetaStore